samples/apps/copilot-chat-app/WebApp/src/components/chat/ChatWindow.tsx

```diff
@@ -65,27 +76,43 @@ const useClasses = makeStyles({
 
 export const ChatWindow: React.FC = () => {
     const classes = useClasses();
+    const sk = useSemanticKernel(process.env.REACT_APP_BACKEND_URI as string);
     const dispatch = useAppDispatch();
     const { conversations, selectedId } = useAppSelector((state: RootState) => state.conversations);
+    const chatName = conversations[selectedId].title;
     const [title, setTitle] = useState<string | undefined>(selectedId ?? undefined);
     const [isEditing, setIsEditing] = useState<boolean>(false);
 
-    const onEdit = () => {
+    const onEdit = async () => {
         if (isEditing) {
-            if (selectedId !== title) 
-                dispatch(editConversationTitle({ id: selectedId ?? '', newId: title ?? ''}));
+            if (chatName !== title) {
+                try {
+                    var ask: IAsk = {
+                        input: conversations[selectedId].id!,
+                        variables: [
+                            { key: 'title', value: title! },
+                        ],
+                    };
+
+                    await sk.invokeAsync(ask, 'ChatMemorySkill', 'EditChat');
+                    dispatch(editConversationTitle({ id: selectedId, newTitle: title ?? '' }));
+                } catch (e: any) {
+                    alert('[EditChat] Unable to retrieve chat. Details:\n' + e);
+                }
+            }
         }
         setIsEditing(!isEditing);
     };
 
     const onTitleChange = (_ev: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
+        console.log(data.value);
```

Comment:
Might want to drop this, unless it helps the developer get familiar with something they are likely to want to change

Reply:
Good catch!